Lib/site.py

```
@@ -168,6 +169,14 @@ def addpackage(sitedir, name, known_paths):
     else:
         reset = False
     fullname = os.path.join(sitedir, name)
+    try:
+        st = os.lstat(fullname)
+    except OSError:
```

Hum, maybe add a _trace() call to help debugging such issue. Something like:

```
_trace(f"Skipping .pth file, stat failed: {exc!r}")
```

---

When `open()` fails below, it is simply silently returns.

If we want to add a `_trace()` for OSError, we should do this in both cases (and maybe in more cases). But this is a different issue. It can add a noise where it was silent.

---

Ah, I didn't notice. In this case, would you mind to add a comment (in the 2 places) to explain that ignoring silently the file on OSError is a deliberate choice?

---

Most `OSError`s are silently ignored in this module and do not have a comment. I do not know what to add in these two places and why they are special. I also not sure that it is a deliberate choice, it can just be a copied behavior.

---

As you want, I already approved the PR.